include/engine/api/route_api.hpp

```diff
@@ -228,42 +228,50 @@ class RouteAPI : public BaseAPI
 
         std::vector<util::json::Object> annotations;
 
-        if (parameters.annotations_type != RouteParameters::AnnotationsType::None)
+        if (parameters.annotations_type != RouteParameters::AnnotationsType::None ||
+            parameters.annotations == true)
         {
+            auto ReqAnnotations = parameters.annotations_type;
+            if ((parameters.annotations == true) & (parameters.annotations_type &
+                RouteParameters::AnnotationsType::None))
+            {
+                ReqAnnotations = RouteParameters::AnnotationsType::All;
+            }
+
             for (const auto idx : util::irange<std::size_t>(0UL, leg_geometries.size()))
             {
                 auto &leg_geometry = leg_geometries[idx];
                 util::json::Object annotation;
 
-                if (parameters.annotations_type & RouteParameters::AnnotationsType::Duration)
+                if (ReqAnnotations & RouteParameters::AnnotationsType::Duration)
```

Comment:
maybe we should add a comment explaining to the future that this basically means
"if Duration bit was set"